cli/internal/providers/event-stream/source/handler.go

```diff
@@ -449,7 +481,128 @@ func (h *Handler) FormatForExport(
 	idNamer namer.Namer,
 	inputResolver resolver.ReferenceResolver,
 ) ([]importremote.FormattableEntity, error) {
-	return nil, fmt.Errorf("formatting event stream sources for export is not supported")
+	sources := collection.GetAll(ResourceType)
+	if len(sources) == 0 {
+		return nil, nil
+	}
+	workspaceMetadata := importremote.WorkspaceImportMetadata{
+		Resources: make([]importremote.ImportIds, 0),
+	}
+	var result []importremote.FormattableEntity
+	for _, source := range sources {
+		data, ok := source.Data.(*sourceClient.EventStreamSource)
+		if !ok {
+			return nil, fmt.Errorf("unable to cast remote resource to event stream source")
+		}
+		workspaceMetadata.WorkspaceID = data.WorkspaceID
+		workspaceMetadata.Resources = []importremote.ImportIds{
+			{
+				LocalID:  source.ExternalID,
+				RemoteID: source.ID,
+			},
+		}
+		spec, err := h.toImportSpec(data, source.ExternalID, workspaceMetadata, collection)
+		if err != nil {
+			return nil, fmt.Errorf("creating spec: %w", err)
+		}
+		result = append(result, importremote.FormattableEntity{
+			Content:      spec,
+			RelativePath: filepath.Join(h.importDir, fmt.Sprintf("%s.yaml", source.ExternalID)),
+		})
+	}
+	return result, nil
+}
+
+func (p *Handler) toImportSpec(source *sourceClient.EventStreamSource, externalID string, workspaceMetadata importremote.WorkspaceImportMetadata, collection *resources.ResourceCollection) (*specs.Spec, error) {
+	metadata := importremote.Metadata{
+		Name: MetadataName,
+		Import: importremote.WorkspacesImportMetadata{
+			Workspaces: []importremote.WorkspaceImportMetadata{workspaceMetadata},
+		},
+	}
+	metadataMap := make(map[string]any)
+	err := mapstructure.Decode(metadata, &metadataMap)
+	if err != nil {
+		return nil, fmt.Errorf("decoding metadata: %w", err)
+	}
+
+	specMap := map[string]any{
+		IDKey:               externalID,
+		NameKey:             source.Name,
+		SourceDefinitionKey: source.Type,
+		EnabledKey:          source.Enabled,
+	}
+
+	if source.TrackingPlan != nil {
+		trackingPlanResource, exists := collection.GetByID(dcstate.TrackingPlanResourceType, source.TrackingPlan.ID)
+		if !exists {
+			return nil, fmt.Errorf("tracking plan with ID %s not found in collection", source.TrackingPlan.ID)
+		}
+
+		validations := map[string]any{
+			TrackingPlanRefYAMLKey:     fmt.Sprintf("#/tracking-plans/%s/%s", dcstate.TrackingPlanResourceType, trackingPlanResource.ExternalID),
+			TrackingPlanConfigYAMLKey: toTrackingPlanConfigImportSpec(source.TrackingPlan.Config),
+		}
+
+		specMap[GovernanceYAMLKey] = map[string]any{
+			ValidationsYAMLKey: validations,
```

Comment:
I got confused earlier about the reference generation and calculation. This is not the correct way, you would need to ask the resolver for fetching the reference of the tracking-plan and not construct yourself.

Because this will be set by the trackingplan resource provider and you would have no control over what it looks like.